test/cases/ostree-vsphere.sh

```diff
@@ -0,0 +1,884 @@
+greenprint "🧹 Remove existing vmdk disk files in vsphere datacenter"
+sudo sshpass -p "${DATACENTER_PASSWORD}" ssh "${SSH_OPTIONS[@]}" "${DATACENTER_USERNAME}@${VSPHERE_DC70_HOST}" \
+    rm -fr /vmfs/volumes/${VSPHERE_DC70_DATASTORE}/edge-vsphere
+
+# Upload image to vsphere and create VM
+export GOVC_URL="https://bootp-73-75-144.lab.eng.pek2.redhat.com/sdk"
+export GOVC_USERNAME="${VSPHERE_USERNAME}"
+export GOVC_PASSWORD="${VSPHERE_PASSWORD}"
+export GOVC_INSECURE="true"
+
+greenprint "📋 Uploading vmdk image to vsphere datacenter 7.0"
+export GOVC_DATACENTER="Datacenter7.0-AMD"
+govc import.vmdk -ds="datastore-21" -pool="/Datacenter7.0-AMD/host/Cluster7.0-AMD/Resources" ./edge-vsphere.vmdk
+
+# Create vm with vmdk image
+greenprint "📋 Create vm in vsphere datacenter"
+sudo ansible-playbook -v -i "${TEMPDIR}"/inventory \
+    -e vsphere_host="${VSPHERE_HOST}" \
+    -e vsphere_username="${VSPHERE_USERNAME}" \
+    -e vsphere_password="${VSPHERE_PASSWORD}" \
+    -e vsphere_datacenter="${VSPHERE_DC70}" \
+    -e vsphere_esxi_host="${VSPHERE_DC70_HOST}" \
+    -e vsphere_datastore="${VSPHERE_DC70_DATASTORE}" \
+    -e vsphere_vm_name="${DC70_VSPHERE_VM_NAME}" \
+    /usr/share/tests/osbuild-composer/ansible/vmdk-create-vm.yaml || RESULTS=0
+
+DC70_GUEST_ADDRESS=$(cat /usr/share/tests/osbuild-composer/ansible/vmdk_ip.txt)
+sudo rm -fr /usr/share/tests/osbuild-composer/ansible/vmdk_ip.txt
+greenprint "🛃 Edge VM IP address is: ${DC70_GUEST_ADDRESS}"
+
+# Run ansible check on edge vm
+# Check for ssh ready to go.
+greenprint "🛃 Checking for SSH is ready to go"
+for _ in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up "${DC70_GUEST_ADDRESS}")"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# With new ostree-libs-2022.6-3, edge vm needs to reboot twice to make the /sysroot readonly
+sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" "${IGNITION_USER}@${DC70_GUEST_ADDRESS}" "echo ${IGNITION_USER_PASSWORD} |nohup sudo -S systemctl reboot &>/dev/null & exit"
+# Sleep 10 seconds here to make sure vm restarted already
+sleep 10
+for _ in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up "${DC70_GUEST_ADDRESS}")"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# Check image installation result
+check_result
+
+greenprint "🕹 Get ostree install commit value"
+INSTALL_HASH=$(curl "${PROD_REPO_URL}/refs/heads/${OSTREE_REF}")
+
+sudo tee "${TEMPDIR}"/inventory > /dev/null << EOF
+[ostree_guest]
+${DC70_GUEST_ADDRESS}
+
+[ostree_guest:vars]
+ansible_python_interpreter=/usr/bin/python3
+ansible_user=${IGNITION_USER}
+ansible_private_key_file=${SSH_KEY}
+ansible_ssh_common_args="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+ansible_become=yes
+ansible_become_method=sudo
+ansible_become_pass=${IGNITION_USER_PASSWORD}
+EOF
+
+# Test IoT/Edge OS
+sudo ansible-playbook -v -i "${TEMPDIR}"/inventory \
```

Comment:
You don't define -e ingition="true", so all ignition test will not be run in check_ostree playbook.

Reply:
Enabled ignition.